lib/Memecoin.ts

```diff
@@ -973,7 +1037,7 @@ export class MemecoinSDK {
       const result = await writeContracts(walletClient, {
         contracts: [
           // approve
-          ...(allowance < amountIn ? [approveContractCall] : []),
+          ...(tokenInPoolType !== TokenPoolType.WETH ? [approveContractCall] : []),
```

Comment:
you still need to check allowance and prevent an extra approveContractCall call. You can do it during estimation phase and pass in the param